tests/test_xliff1.py

```diff
@@ -0,0 +1,718 @@
+# Copyright Mozilla Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from importlib.resources import files
+from textwrap import dedent
+from unittest import TestCase
+
+from moz.l10n.message import (
+    CatchallKey,
+    Expression,
+    FunctionAnnotation,
+    Markup,
+    PatternMessage,
+    SelectMessage,
+    VariableRef,
+)
+from moz.l10n.resource import Comment, Entry, Metadata, Resource, Section
+from moz.l10n.xliff import xliff_parse, xliff_serialize
+
+# Show full diff in self.assertEqual. https://stackoverflow.com/a/61345284
+# __import__("sys").modules["unittest.util"]._MAX_LENGTH = 999999999
+
+hello = files("tests.data").joinpath("hello.xliff").read_bytes()
+angular = files("tests.data").joinpath("angular.xliff").read_bytes()
+icu_docs = files("tests.data").joinpath("icu-docs.xliff").read_bytes()
+xcode = files("tests.data").joinpath("xcode.xliff").read_bytes()
+
+
+class TestXliff1(TestCase):
```

Comment:
Why `TestXliff1` rather than `TestXliff`? The same goes for the file name.

Reply:
Because this is testing XLIFF 1 resources only. If/once we find a reason to support XLIFF 2, that can be added within the existing `xliff_parse()` & `xliff_serialize()` APIs. The test suites aren't a part of the public API, so here being explicit isn't a problem.